python/hail/linalg/blockmatrix.py

```diff
@@ -128,15 +128,23 @@ def from_entry_expr(cls, entry_expr, path=None, block_size=None):
             path = new_temp_file(suffix="bm")
         if not block_size:
             block_size = cls.default_block_size()
+
         source = entry_expr._indices.source
         if not isinstance(source, MatrixTable):
             raise ValueError("Expect an expression of 'MatrixTable', found {}".format(
                 "expression of '{}'".format(source.__class__) if source is not None else 'scalar expression'))
-        mt = source
-        base, _ = mt._process_joins(entry_expr)
-        analyze('from_entry_expr', entry_expr, mt._entry_indices)
 
-        base._jvds.writeBlockMatrix(path, to_expr(entry_expr)._ast.to_hql(), block_size)
+        if entry_expr._indices != source._entry_indices:
```

Comment:
code like this is duplicated in a bunch of places. I wonder what the best way to abstract it is -- maybe add a private method _to_matrix_table on Expression which does something similar to _to_table? not necessary here, but let's think about that going forward.